src/DotNetty.Transport/Channels/Sockets/SocketDatagramChannel.cs

```diff
@@ -0,0 +1,479 @@
+            }
+            finally
+            {
+                if (!success)
+                {
+                    this.DoClose();
+                }
+            }
+        }
+
+        protected override void DoFinishConnect(SocketChannelAsyncOperation operation)
+        {
+            throw new NotSupportedException();
+        }
+
+        protected override void DoDisconnect() => this.DoClose();
+
+        protected override void DoClose()
+        {
+            if (this.TryResetState(StateFlags.Open | StateFlags.Active))
+            {
+                this.Socket.Shutdown(SocketShutdown.Both);
+                this.Socket.Dispose();
+            }
+        }
+
+        protected override void ScheduleSocketRead()
+        {
+            SocketChannelAsyncOperation operation = this.ReadOperation;
+            operation.RemoteEndPoint = this.anyRemoteEndPoint;
+
+            IRecvByteBufAllocatorHandle handle = this.Unsafe.RecvBufAllocHandle;
+            IByteBuffer buffer = handle.Allocate(this.config.Allocator);
+            handle.AttemptedBytesRead = buffer.WritableBytes;
+            operation.UserToken = buffer;
+
+            ArraySegment<byte> bytes = buffer.GetIoBuffer();
+            operation.SetBuffer(bytes.Array, bytes.Offset, buffer.WritableBytes);
+
+            bool pending;
+#if NETSTANDARD1_3
+            pending = this.Socket.ReceiveFromAsync(operation);
+#else
+            if (ExecutionContext.IsFlowSuppressed())
+            {
+                pending = this.Socket.ReceiveFromAsync(operation);
+            }
+            else
+            {
+                using (ExecutionContext.SuppressFlow())
+                {
+                    pending = this.Socket.ReceiveFromAsync(operation);
+                }
+            }
+#endif
+            if (!pending)
+            {
+                this.EventLoop.Execute(ReceiveFromCompletedSyncCallback, this.Unsafe, operation);
+            }
+        }
+
+        protected override int DoReadMessages(List<object> buf)
+        {
+            Contract.Requires(buf != null);
+
+            SocketChannelAsyncOperation operation = this.ReadOperation;
+
+            var data = (IByteBuffer)operation.UserToken;
+            bool free = true;
+            try
+            {
+                int received = operation.BytesTransferred;
+                if (received <= 0)
+                {
+                    return 0;
+                }
+
+                IRecvByteBufAllocatorHandle handle = this.Unsafe.RecvBufAllocHandle;
+                handle.LastBytesRead = received;
+                data.SetWriterIndex(data.WriterIndex + received);
+                EndPoint remoteAddress = operation.RemoteEndPoint;
+                buf.Add(new DatagramPacket(data, remoteAddress, this.LocalAddress));
+                free = false;
+
+                return 1;
+            }
+            catch (Exception exception)
+            {
+                Logger.Error(exception);
```

Comment:
`PlatformDependent.throwException(cause);` in netty here will actually just throw. It's a way for them to avoid specifying `throws ...` on the method while it may actually throw.
Just remove the whole `catch` block pls.